flink/v1.17/flink/src/test/java/org/apache/iceberg/flink/source/TestFlinkMergingMetrics.java

```diff
@@ -18,51 +18,47 @@
  */
 package org.apache.iceberg.flink.source;
 
+import java.io.File;
 import java.io.IOException;
 import java.util.List;
 import org.apache.flink.table.data.RowData;
 import org.apache.flink.table.types.logical.RowType;
-import org.apache.iceberg.FileFormat;
 import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Table;
 import org.apache.iceberg.TestMergingMetrics;
 import org.apache.iceberg.data.Record;
 import org.apache.iceberg.flink.FlinkSchemaUtil;
-import org.apache.iceberg.flink.HadoopTableResource;
+import org.apache.iceberg.flink.HadoopCatalogExtension;
 import org.apache.iceberg.flink.RowDataConverter;
+import org.apache.iceberg.flink.TestFixtures;
 import org.apache.iceberg.flink.sink.FlinkAppenderFactory;
 import org.apache.iceberg.io.FileAppender;
 import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
-import org.junit.ClassRule;
-import org.junit.Rule;
-import org.junit.rules.TemporaryFolder;
+import org.junit.jupiter.api.extension.RegisterExtension;
 
 public class TestFlinkMergingMetrics extends TestMergingMetrics<RowData> {
 
-  @ClassRule public static final TemporaryFolder TEMP_FOLDER = new TemporaryFolder();
-
-  @Rule
-  public final HadoopTableResource tableResource =
-      new HadoopTableResource(TEMP_FOLDER, "test_db", "test_table", SCHEMA);
-
-  public TestFlinkMergingMetrics(FileFormat fileFormat) {
-    super(fileFormat);
-  }
+  @RegisterExtension
+  private static final HadoopCatalogExtension catalogExtension =
```

Comment:
can you please make sure that this static final is named in UPPERCASE across the PR? I've just realized that this isn't the case for existing code, wh we should eventually fix and enforce via checkstyle (in case you're interested in contributing and working on this)

Reply:
Update: fixed one instance of this after having double checked, pushed the commit.